pkg/driver/bacnet/merge/emergency.go

```diff
@@ -100,6 +103,41 @@ func (t *emergencyImpl) PullEmergency(request *traits.PullEmergencyRequest, serv
 	return t.EmergencyApiServer.PullEmergency(request, server)
 }
 
+func (t *emergencyImpl) checkIntValueForEmergency(response any) (*traits.Emergency, error) {
+	data := &traits.Emergency{}
+
+	value, err := comm.IntValue(response)
+	if err != nil {
+		return nil, comm.ErrReadProperty{Prop: "alarmConfig", Cause: err}
+	}
+
+	if value != int64(t.config.AlarmConfig.OkValue) ||
+		value < int64(t.config.AlarmConfig.GreaterThanOkValue) {
+		data.Reason = t.config.AlarmConfig.AlarmReason
+		data.Level = traits.Emergency_EMERGENCY
+	} else {
```

Comment:
See comment on the config structure, this condition incorrect results in error due to default/zero values of either configured condition.